exercises/allergies/src/Example.hx

```diff
@@ -0,0 +1,28 @@
+package;
+
+enum Allergen {
+    Eggs;
+    Peanuts;
+    Shellfish;
+    Strawberries;
+    Tomatoes;
+    Chocolate;
+    Pollen;
+    Cats;
+}
+
+class Allergies {
+    private final score: EnumFlags<Allergen>;
+
+    public function new(score: Int) {
+        this.score = new EnumFlags<Allergen>(score);
```

Comment:
I'd like to avoid using class constructors and stick with static fields.

Reply:
Do you mean using class properties instead of constructors? Score here can't be made static as it's per-instance.

Reply:
I would like something along these lines:

```haxe
package;

class Allergies {
    public static function allergicTo(item: String, score: Int): Bool {
        throw "Not Implemented"; // Delete this statement and write your own implementation.
    }

    public static function list(score: Int): Array<String> {
        throw "Not Implemented"; // Delete this statement and write your own implementation.
    } 
}
```

allergicTo test example:

```haxe
it("allergic to everything", {
    pending("Skipping");
    Allergies.allergicTo("shellfish", 255).should.be(true);
});
```

list test example:
```haxe
it("lots of stuff", {
    pending("Skipping");
    Allergies.list(248).should.containExactly([
        "strawberries",
        "tomatoes",
        "chocolate",
        "pollen",
        "cats"
    ]);
});
```

You don't have to use this, but some here's some inspiration for something similar to your code:
```haxe
package;

import haxe.EnumFlags;

using StringTools;

enum Allergen {
    Eggs;
    Peanuts;
    Shellfish;
    Strawberries;
    Tomatoes;
    Chocolate;
    Pollen;
    Cats;
}

class Test {
    public static function main() {
        var result = new Array<String>();
        var allergies = new EnumFlags<Allergen>(248);
        for (type in Type.allEnums(Allergen)) 
            if (allergies.has(type))
                 result.push('${type}'.toLowerCase());
        
        trace(result);
       //[strawberries,tomatoes,chocolate,pollen,cats]
    }
}
```

I hope this helps :)

Reply:
Ah, got what you mean now.
Thanks for the sample code. That's pretty clear.